switools/swisignature.py

```diff
@@ -144,9 +160,123 @@ def signSwiHandler( args ):
    rootCaFile = args.CAfile
    signatureFile = args.signature
    signingKeyFile = args.key
-   signSwi( swi, signingCertFile, rootCaFile, signatureFile, signingKeyFile )
+   signSwiAll( swi, signingCertFile, rootCaFile, signatureFile, signingKeyFile )
    print( 'SWI/X file %s successfully signed and verified.' % swi )
 
+def extractSignature( swi, destFile, sigFileName=None ):
+   sigFn = sigFileName
+   if not sigFn:
+      sigFn = signaturelib.SWI_SIG_FILE_NAME
+   ret = os.system( "set -e;"
+                    "destDir=$(readlink -f $(dirname %s));"
+                    "destFile=$(basename %s);"
+                    "swi=$(readlink -f %s);"
+                    "sigFile=%s;"
+                    "cd $destDir;"
+                    "unzip -o -q $swi $sigFile;"
+                    "mv $sigFile $destFile" % (
+                      destFile, destFile, swi, sigFn )
+                  )
+   if ret != 0:
+      print( "Error: Cannot extract signature file %s from  %s" % ( sigFn, swi ) )
+      sys.exit( SWI_SIGN_RESULT.CANNOT_EXTRACT_SIGNATURE )
+
+def getSignatureFile( swi, signatureFile ):
+   serviceName = 'swi-signing-service'
+   serviceBinary = shutil.which( serviceName )
+   if not serviceBinary:
+      print( "Error: signing service '%s' not found" % serviceName )
+      sys.exit( SWI_SIGN_RESULT.ERROR_SIGNING_SERVER_FAILED )
+
+   sha256 = prepareSwi( swi=swi, outfile=None, forceSign=True )
+   print( "%s sha256: %s" % ( os.path.basename( swi ), sha256 ) )
+   ret = subprocess.check_call( [ 'swi-signing-service', sha256, signatureFile ] )
+   if ret != 0:
```

Comment:
`subprocess.check_call` doesn't return anything, but instead it raises `subprocess.CalledProcessError` if the command fails.
```suggestion
   try:
      subprocess.check_call( [ 'swi-signing-service', sha256, signatureFile ] )
   except subprocess.CalledProcessError:
```